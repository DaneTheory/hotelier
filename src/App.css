.window {
  border: 1px solid #d2d2d2;
}