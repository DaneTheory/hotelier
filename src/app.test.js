import React from 'react';
import App from './app';
import renderer from 'react-test-renderer';

import mockFetch from 'jest-fetch-mock';

describe('App', () => {
  function mockSuccessfulResponse() {
    mockFetch.mockResponse('{"server1":{"status":"running"},"server2":{"status":"stopped"}}');
  }

  it('matches the snapshot', done => {
    mockSuccessfulResponse();
    const rendered = renderer.create(<App/>);
    expect(rendered.toJSON()).toMatchSnapshot('loading');
    process.nextTick(() => {
      expect(rendered.toJSON()).toMatchSnapshot('loaded');
      done();
    });
  });

  it('updates the state with the servers loaded from the API', done => {
    mockSuccessfulResponse();
    const instance = renderer.create(<App/>).getInstance();
    expect(instance.state.loading).toBe(true);
    process.nextTick(() => {
      expect(instance.state.servers.length).toEqual(2);
      expect(instance.state.servers[0].id).toEqual('server1');
      expect(instance.state.servers[0].status).toEqual('running');
      expect(instance.state.servers[1].id).toEqual('server2');
      expect(instance.state.servers[1].status).toEqual('stopped');
      expect(instance.state.loading).toBe(false);
      done();
    });
  });

  it('renders empty list when we get error from the api', done => {
    try {
      mockFetch.mockResponse('', { status: 500 });
      const instance = renderer.create(<App/>).getInstance();
      process.nextTick(() => {
        expect(instance.state.servers).toEqual([]);
        done();
      });
    } catch (e) {
      done();
    }
  });

  describe('#updateServerStatus', () => {
    beforeEach(() => mockSuccessfulResponse());

    it('updates the corresponding server with the new status', done => {
      const instance = renderer.create(<App/>).getInstance();
      process.nextTick(() => {
        instance.updateServerStatus('server1', 'stopped');
        expect(instance.state.servers[0].status).toEqual('stopped');
        done();
      });
    });
  });
});
