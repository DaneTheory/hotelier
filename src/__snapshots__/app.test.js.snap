// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loaded 1`] = `
<div>
  <div
    className="header-arrow"
  />
  <div
    className="window"
  >
    <header
      className="toolbar toolbar-header"
    >
      <h1
        className="title"
      >
        Hotelier
      </h1>
    </header>
    <div
      className="window-content"
    >
      <div
        className="pane"
      >
        <div>
          <table>
            <thead>
              <tr>
                <th>
                  Server
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  onClick={[Function]}
                >
                  server1
                </td>
                <td>
                  <button
                    className="btn btn-negative"
                    onClick={[Function]}
                  >
                    Stop
                  </button>
                </td>
              </tr>
              <tr>
                <td
                  onClick={[Function]}
                >
                  server2
                </td>
                <td>
                  <button
                    className="btn btn-positive"
                    onClick={[Function]}
                  >
                    Start
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <footer
      className="toolbar toolbar-footer"
    >
      <div
        className="toolbar-actions pull-left"
      >
        <button
          className="btn btn-default"
          onClick={[Function]}
        >
          Hotel
        </button>
      </div>
      <div
        className="toolbar-actions pull-right"
      >
        <div
          className="btn-group"
        >
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            <span
              className="icon icon-arrows-ccw"
              title="Refresh"
            />
          </button>
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            <span
              className="icon icon-cancel"
              title="Quit"
            />
          </button>
        </div>
      </div>
    </footer>
  </div>
</div>
`;

exports[`loading 1`] = `
<div>
  <div
    className="header-arrow"
  />
  <div
    className="window"
  >
    <header
      className="toolbar toolbar-header"
    >
      <h1
        className="title"
      >
        Hotelier
      </h1>
    </header>
    <div
      className="window-content"
    >
      <div
        className="pane"
      >
        <div
          className="summary"
        >
          Loadingâ€¦
        </div>
        <div>
          <table>
            <thead>
              <tr>
                <th>
                  Server
                </th>
                <th>
                  Action
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
    <footer
      className="toolbar toolbar-footer"
    >
      <div
        className="toolbar-actions pull-left"
      >
        <button
          className="btn btn-default"
          onClick={[Function]}
        >
          Hotel
        </button>
      </div>
      <div
        className="toolbar-actions pull-right"
      >
        <div
          className="btn-group"
        >
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            <span
              className="icon icon-arrows-ccw"
              title="Refresh"
            />
          </button>
          <button
            className="btn btn-default"
            onClick={[Function]}
          >
            <span
              className="icon icon-cancel"
              title="Quit"
            />
          </button>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
